dfusion_rust_core/src/models/order.rs

```diff
@@ -47,6 +47,73 @@ impl Order {
             sell_amount,
         }
     }
+
+    /// Calculates the utility of an order based on a buy and sell price as well
+    /// as the executed buy amount.
+    ///
+    /// # Returns
+    ///
+    /// Returns utility of an order or None when there was an error in the
+    /// calculation such as overflow or divide by 0.
+    pub fn utility(
+        &self,
+        buy_price: u128,
+        exec_buy_amount: u128,
+        exec_sell_amount: u128,
+    ) -> Option<U256> {
+        let buy_price = u128_to_u256(buy_price);
+        let exec_buy_amount = u128_to_u256(exec_buy_amount);
+        let exec_sell_amount = u128_to_u256(exec_sell_amount);
+        let buy_amount = u128_to_u256(self.buy_amount);
+        let sell_amount = u128_to_u256(self.sell_amount);
+
+        // essential_utility = (exec_buy_amount - (exec_sell_amount * buy_amount)
+        //   / sell_amount) * buy_price
+        // essential_utility = (((exec_sell_amount * buy_amount) % sell_amount)
```

Comment:
These comments don't match up 

`utility = essential_utility - essential_utility`

==> utility = 0

Is there any rusty way to make sure those comments are up to date? E.g. Using a documentation test or such?